contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+```yaml
+Kind: SchedulingPolicy
+spec:
+  required:
+    SchedulerNames: ["green-scheduler", "my-scheduler"]
+```
+
+An empty list of schedulerNames here is not a valid syntax:
+```yaml
+Kind: SchedulingPolicy
+spec:
+  required:
+    SchedulerNames: [] # equivalent to ["default-scheduler"] or to not specifying this item
+```
+
+
+#### allowed
+Allow pods to use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`):
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    SchedulerNames: ["green-scheduler", "my-scheduler"]
+```
+
+An empty list of schedulerNames will allow usage of all schedulers:
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    SchedulerNames: []
+```
+
+#### default
+pods will default use either the my-scheduler if nothing is specified in `spec.schedulerName`:
+```yaml
+Kind: SchedulingPolicy
+spec:
+  default:
+    SchedulerName: "my-scheduler"
+```
+
+
+### Tolerations
+
+Toleration usage can be regulated using fine-grain rules with `tolerations` field. If specifying multiple `tolerations`, pod will be scheduled if one of the tolerations is satisfied.
+
+
+#### Allowed
+
+This allows requires tolerations in the following forms:
+- tolerations that tolerates taints with key named `mykey` that has a value `value` and with a `NoSchedule` effect.
+- tolerations that tolerates taints with key `other_key` that has a `NoExecute` effect.
+
+##### Fine-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: ["mykey"]
+      operators: ["Equal"]
+      values: ["value"]
+      effects: ["NoSchedule"]
+    - keys: ["other_key"]
+      operators: ["Exists"]
+      effects: ["NoExecute"]
+```
+
+Here we allow tolerations in the following forms:
+- tolerations that tolerates all `PreferNoSchedule` taints with any value.
+- tolerations that tolerates taints based on any key existence with effect `NoSchedule`.
+Also note that this SchedulingPolicy does not allow tolerating NoExecute taints.
+
+##### Coarse-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: ["PreferNoSchedule"]
+    - keys: []
+      operators: ["Exists"]
+      effects: ["NoSchedule"]
+```
+an empty list of toleration allows all types of tolerations:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations: []
+```
+
+Which is equivalent to:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: []
+```
+
+#### default
+
+if no toleration is not specified, the following `SchedulingPolicy` will add a:
```

Comment:
Defaulting in Kubernetes is sort of a mess. Defaulting tolerations is going to be tricky, since it means there's no way to specify that a pod shouldn't have any tolerations. I.e. we don't distinguish between an empty list, and an unset list.

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required

The list needs to be optional.